File: freecodecamp/daily-code-challenge/2025-09-29.js
/*

Longest Word

Given a sentence, return the longest word in the sentence.
- Ignore periods (.) when determining word length.
- If multiple words are ties for the longest, return the first one that occurs.


Tests : 
1. getLongestWord("coding is fun") should return "coding".
2. getLongestWord("Coding challenges are fun and educational.") should return "educational".
3. getLongestWord("This sentence has multiple long words.") should return "sentence".

*/

function getLongestWord(sentence) {
  if (!sentence || typeof sentence !== 'string') {
    return '';
  }
  
  const words = sentence
    .replace(/\./g, '')
    .split(" ")
    .filter(word => word.length > 0);
  
  if (words.length === 0) {
    return '';
  }
  
  return words.reduce((longest, current) => 
    current.length > longest.length ? current : longest
  );
}

// Brute Force

function getLongestWord(sentence) {
  const newSentence = sentence.replace(".", "");
  let words = newSentence.split(" ");

  let longestWordLength = 0;
  let longestWord = '';

  for(let i=0; i < words.length; i++) {
    let len = words[i].length;
    if(longestWordLength < words[i].length) {
      longestWordLength = len;
      longestWord = words[i];
    }
  }

  return longestWord;
}

File: freecodecamp/daily-code-challenge/2025-09-29.py
"""

Longest Word

Given a sentence, return the longest word in the sentence.
- Ignore periods (.) when determining word length.
- If multiple words are ties for the longest, return the first one that occurs.


Examples : 
1. getLongestWord("coding is fun") should return "coding".
2. getLongestWord("Coding challenges are fun and educational.") should return "educational".
3. getLongestWord("This sentence has multiple long words.") should return "sentence".

"""

def get_longest_word(sentence):
    if not sentence or not isinstance(sentence, str):
        return ''
    
    words = sentence.replace('.', '').split()
    
    if not words:
        return ''
    
    return max(words, key=len)

# Brute Force

def get_longest_word(sentence):
    if not sentence or not isinstance(sentence, str):
        return ''
    
    # Remove periods
    sentence = sentence.replace('.', '')
    
    # Split into words
    words = sentence.split()
    
    if not words:
        return ''
    
    # Brute force: iterate through all words
    longest_word = ''
    longest_length = 0
    
    for word in words:
        word_length = len(word)
        if word_length > longest_length:
            longest_length = word_length
            longest_word = word
    
    return longest_word
